nf_core/subworkflows/lint/subworkflow_tests.py

```diff
@@ -0,0 +1,119 @@
+"""
+Lint the tests of a subworkflow in nf-core/modules
+"""
+import logging
+import os
+
+import yaml
+
+import nf_core.subworkflows
+
+log = logging.getLogger(__name__)
+
+
+def subworkflow_tests(_, subworkflow):
+    """
+    Lint the tests of a subworkflow in ``nf-core/modules``
+
+    It verifies that the test directory exists
+    and contains a ``main.nf`` and a ``test.yml``,
+    and that the subworkflow is present in the ``pytest_modules.yml``
+    file.
+
+    Additionally, hecks that all included components in test ``main.nf`` are specified in ``test.yml``
+    """
+
+    if os.path.exists(subworkflow.test_dir):
+        subworkflow.passed.append(("test_dir_exists", "Test directory exists", subworkflow.test_dir))
+    else:
+        subworkflow.failed.append(("test_dir_exists", "Test directory is missing", subworkflow.test_dir))
+        return
+
+    # Lint the test main.nf file
+    test_main_nf = os.path.join(subworkflow.test_dir, "main.nf")
+    if os.path.exists(test_main_nf):
+        subworkflow.passed.append(("test_main_exists", "test `main.nf` exists", subworkflow.test_main_nf))
+    else:
+        subworkflow.failed.append(("test_main_exists", "test `main.nf` does not exist", subworkflow.test_main_nf))
+
+    # Check that entry in pytest_modules.yml exists
+    try:
+        pytest_yml_path = os.path.join(subworkflow.base_dir, "tests", "config", "pytest_modules.yml")
+        with open(pytest_yml_path, "r") as fh:
+            pytest_yml = yaml.safe_load(fh)
+            if "subworkflows/" + subworkflow.component_name in pytest_yml.keys():
+                subworkflow.passed.append(("test_pytest_yml", "correct entry in pytest_modules.yml", pytest_yml_path))
+            else:
+                subworkflow.failed.append(("test_pytest_yml", "missing entry in pytest_modules.yml", pytest_yml_path))
+    except FileNotFoundError:
+        subworkflow.failed.append(("test_pytest_yml", "Could not open pytest_modules.yml file", pytest_yml_path))
+
+    # Lint the test.yml file
+    try:
+        with open(subworkflow.test_yml, "r") as fh:
+            test_yml = yaml.safe_load(fh)
+
+            # Verify that tags are correct
```

Comment:
Similar to my previous comment in `meta_yml.py`, currently the `test.yml` file lists all included modules and subworkflows (required for pystests to run). This test validates that `test.yml` includes at least these components.
In most of the subworkflows in nf-core/modules, we also list modules not directly included in the current subworkflow but in other included subworkflows. This is more difficult to lint, but probably good to have, as pytest won't run for these modules otherwise.